crates/core/src/doc/field.rs

```diff
@@ -1,24 +1,40 @@
 use crate::ctx::{Context, MutableContext};
-use crate::dbs::Options;
-use crate::dbs::Statement;
 use crate::dbs::capabilities::ExperimentalTarget;
+use crate::dbs::{Options, Statement};
 use crate::doc::Document;
 use crate::err::Error;
 use crate::expr::data::Data;
 use crate::expr::idiom::{Idiom, IdiomTrie, IdiomTrieContains};
 use crate::expr::kind::Kind;
 use crate::expr::permission::Permission;
-use crate::expr::reference::Refs;
 use crate::expr::statements::DefineFieldStatement;
-use crate::expr::thing::Thing;
-use crate::expr::value::every::ArrayBehaviour;
-use crate::expr::value::{CoerceError, Value};
+use crate::expr::statements::define::DefineDefault;
 use crate::expr::{FlowResultExt as _, Part};
 use crate::iam::Action;
+use crate::val::value::CoerceError;
+use crate::val::value::every::ArrayBehaviour;
+use crate::val::{RecordId, Value};
 use anyhow::{Result, bail, ensure};
 use reblessive::tree::Stk;
 use std::sync::Arc;
 
+fn clean_none(v: &mut Value) -> bool {
```

Comment:
This was a previous bug so I'd just like to explicitely describe the behaviour of this function. We can probably simplify the `every` function too
```suggestion
/// Removes `NONE` values recursively from objects, but not when `NONE` is a direct child of an array
fn clean_none(v: &mut Value) -> bool {
```